v19.1/build-a-clojure-app-with-cockroachdb.md

```diff
@@ -91,6 +107,100 @@ lein run
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach sql --insecure \
+--database=bank \
+--user=maxroach \
+-e 'CREATE TABLE accounts (id INT PRIMARY KEY, balance INT)'
+~~~
+
+## Step 4. Run the Clojure code
+
+Now that you have a database and a user, you'll run code to create a table and insert some rows, and then you'll run code to read and update values as an atomic [transaction](transactions.html).
+
+### Create a basic Clojure/JDBC project
+
+1. Create a new directory `myapp`.
+2. Create a file `myapp/project.clj` and populate it with the following code, or <a href="https://raw.githubusercontent.com/cockroachdb/docs/master/_includes/{{ page.version.version }}/app/project.clj" download>download it directly</a>.
+
+    {% include copy-clipboard.html %}
+    ~~~ clojure
+    {% include {{ page.version.version }}/app/project.clj %}
+    ~~~
+
+3. Create a file `myapp/src/test/util.clj` and populate it with the code from <a href="https://raw.githubusercontent.com/cockroachdb/docs/master/_includes/{{ page.version.version }}/app/util.clj" download>this file</a>. Be sure to place the file in the subdirectory `src/test` in your project.
+
+### Basic statements
+
+First, use the following code to connect as the `maxroach` user and execute some basic SQL statements, inserting rows and reading and printing the rows.
+
+Create a file `myapp/src/test/test.clj` and copy the code below to it, or <a href="https://raw.githubusercontent.com/cockroachdb/docs/master/_includes/{{ page.version.version }}/app/insecure/basic-sample.clj" download>download it directly</a>. Be sure to rename this file to `test.clj` in the subdirectory `src/test` in your project.
+
+{% include copy-clipboard.html %}
+~~~ clojure
+{% include {{ page.version.version }}/app/insecure/basic-sample.clj %}
+~~~
+
+Run with:
+
+{% include copy-clipboard.html %}
+~~~ shell
+lein run
+~~~
+
+### Transaction (with retry logic)
+
+Next, use the following code to again connect as the `maxroach` user but this time execute a batch of statements as an atomic transaction to transfer funds from one account to another, where all included statements are either committed or aborted.
+
+Copy the code below to `myapp/src/test/test.clj` or
+<a href="https://raw.githubusercontent.com/cockroachdb/docs/master/_includes/{{ page.version.version }}/app/insecure/txn-sample.clj" download>download it directly</a>. Again, preserve the file name `test.clj`.
+{{site.data.alerts.callout_info}}
+CockroachDB may require the
+[client to retry a transaction](transactions.html#transaction-retries) in case of read/write contention. CockroachDB provides a generic **retry function** that runs inside a transaction and retries it as needed. You can copy and paste the retry function from here into your code.
+{{site.data.alerts.end}}
+
+{% include copy-clipboard.html %}
+~~~ clojure
+{% include {{ page.version.version }}/app/insecure/txn-sample.clj %}
+~~~
+
+Run with:
+
+{% include copy-clipboard.html %}
+~~~ shell
+lein run
```

Comment:
We prepend shell commands with `$` (the copy-paste button automagically removes them)

Reply:
Ah OK. I will add those back in.